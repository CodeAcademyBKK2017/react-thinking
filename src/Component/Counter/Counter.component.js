import React, { Component } from 'react';
import CountStatus from '../CountStatus/CountStatus.conmponent';
import Button from '../Button/Button.component';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';

class Counter extends Component {
    render() {
        const {value, add, subtract} = this.props;
        return (
            <div className="App">
                <CountStatus value={value}/>
                <Button func={add} text={'Add'}/>
                <Button func={subtract} text={'Subtract'}/>
            </div>
        );
    }
}
  
Counter.propTypes = {
    add : PropTypes.func.isRequired,
    subtract : PropTypes.func.isRequired
};

Counter.defaultProps = {
    add: noop,
    subtract : noop,
}

export default Counter;