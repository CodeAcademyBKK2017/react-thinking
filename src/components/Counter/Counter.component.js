import React, {Component} from 'react';
import './Counter.style.css';
import CountStatus from '../CountStatus/CountStatus.component';
import Button from '../Button/Button.component';

class Counter extends Component {
	constructor (props) {
		super(props);
		this.state = {
			count: 0
		};
	}

	add = () => {
		this.setState({count: this.state.count+1});
	}

	subtract = () => {
		this.setState({count: this.state.count-1});
	}

	render () {
		return (
			<div className="Counter">
				<CountStatus status={this.state.count} />
				<Button title="Add" type="ADD" onClick={this.add} />
				<Button title="Subtract" type="SUB" onClick={this.subtract} />
			</div>
		);
	}
}

export default Counter;
