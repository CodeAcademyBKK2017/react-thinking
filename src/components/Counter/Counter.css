.Counter {
    max-width: 600px;
    min-width: 450px;

    margin: 50px auto;
    padding: 30px;

    text-align: center;
}