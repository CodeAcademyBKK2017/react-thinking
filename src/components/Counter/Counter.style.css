.Counter{
    align-content: center;
    display: block;
}