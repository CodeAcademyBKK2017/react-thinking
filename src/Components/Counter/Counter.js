import React, { Component } from 'react';
import './Counter.css';

import CountStatus from '../CountStatus/CountStatus';
import Button from '../Button/Button';

class Counter extends Component {
    state = {
        count: 0
    };
    Btn_Add=()=>{
        this.setState({count: this.state.count+1});
    }
    Btn_Sub=()=>{
        this.setState({count: this.state.count-1});
    }
    render() {
        return (
            <div className='counter'>
                <div>
                    <CountStatus count={this.state.count} />
                </div>
                <div align='center'>
                    <Button type='add' title='Add' onclick={this.Btn_Add} /> <Button type='subtract' title='Subtract'  onclick={this.Btn_Sub} />
                </div>
            </div>
        );
    }
}

export default Counter;