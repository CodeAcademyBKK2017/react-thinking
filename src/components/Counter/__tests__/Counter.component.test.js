import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Counter from '../Counter.component';

describe('Counter component', () => {
  it('Counter: should increment state', () => {
    const wrapper = shallow(<Counter />);
    const instance = wrapper.instance();
    expect(instance.state.count).toBe(0);
    instance.add();
    expect(instance.state.count).toBe(1);
  });
  it('Counter: should decrement state', () => {
    const wrapper = shallow(<Counter />);
    const instance = wrapper.instance();
    expect(instance.state.count).toBe(0);
    instance.subtract();
    expect(instance.state.count).toBe(-1);
  });
});