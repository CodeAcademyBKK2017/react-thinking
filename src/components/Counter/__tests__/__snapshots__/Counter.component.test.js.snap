// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Counter: renders correctly 1`] = `
<div
  className="Counter"
>
  <div
    className="CountStatus"
  >
    0
  </div>
  <button
    className="Button"
    onClick={[Function]}
  >
    Add
  </button>
  <button
    className="Button"
    onClick={[Function]}
  >
    Subtract
  </button>
</div>
`;
